app/src/main/java/org/mozilla/fenix/settings/QuickSettingsSheetDialogFragment.kt

```diff
@@ -0,0 +1,265 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import mozilla.components.support.ktx.android.net.hostWithoutCommonPrefixes
+import mozilla.components.support.ktx.kotlin.toUri
+import org.mozilla.fenix.R
+import org.mozilla.fenix.settings.PhoneFeature.NOTIFICATION
+import org.mozilla.fenix.settings.PhoneFeature.LOCATION
+import org.mozilla.fenix.settings.PhoneFeature.CAMERA
+import org.mozilla.fenix.settings.PhoneFeature.MICROPHONE
+import org.mozilla.fenix.utils.Settings
+import java.lang.IllegalArgumentException
+import android.util.TypedValue
+import mozilla.components.feature.sitepermissions.SitePermissions
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage
+import org.jetbrains.anko.textColorResource
+import org.mozilla.fenix.utils.ItsNotBrokenSnack
+
+private const val KEY_URL = "KEY_URL"
+private const val KEY_IS_SECURED = "KEY_IS_SECURED"
+private const val KEY_IS_SITE_IN_EXCEPTION_LIST = "KEY_IS_SITE_IN_EXCEPTION_LIST"
+private const val REQUEST_CODE_QUICK_SETTINGS_PERMISSIONS = 4
+
+@SuppressWarnings("TooManyFunctions")
+class QuickSettingsSheetDialogFragment : BottomSheetDialogFragment() {
```

Comment:
There's a good bit of complexity in this Fragment and it takes some effort to read. Hence the "TooManyFunctions" lint warning. We'd normally break out the View into a UIComponent and UIView in non-trivial cases like this, leaving the Fragment only managing setup, lifecycle, and how it responds to events from the View. Do you think you could break this class out?

Reply:
Sure no problem!

Reply:
Updated!